### PR TITLE
Added time-series plot of integral along cross-section.

### PR DESCRIPTION
**Pull request by varmar05**

### CODE REVIEW

File: crayfish/plot.py [hunk 1]
```diff
@@ -94,6 +95,52 @@ def cross_section_plot_data(layer, ds_group_index, ds_index, geometry, resolutio
...
+
+        _x, _y = [], []
+        integral = 0
+        if not layer:
+            return x, y
+
+        dataset = QgsMeshDatasetIndex(ds_group_index, i)
+        offset = 0
+        length = geometry.length()
+        while offset < length:
```
Comment (PeterPetrik) [inline crayfish/plot.py]: do () while maybe?

File: crayfish/plot.py [hunk 2]
```diff
@@ -94,6 +95,52 @@ def cross_section_plot_data(layer, ds_group_index, ds_index, geometry, resolutio
...
+            pt = geometry.interpolate(offset).asPoint()
+            value = layer.datasetValue(dataset, pt).scalar()
+            # calculate parts when f(x) is continuous
+            if math.isnan(value):
+                if len(_y):
+                    # approximated integral(a,b) f(x) as (b-a)/N sum(0,N) f(x_n)
+                    integral += sum(_y)*(_x[-1]-_x[0])/len(_y)
```
Comment (PeterPetrik) [inline crayfish/plot.py]: this assumes that time intervals are regular?

File: crayfish/plot.py [hunk 3]
```diff
@@ -94,6 +95,52 @@ def cross_section_plot_data(layer, ds_group_index, ds_index, geometry, resolutio
...
+        if not math.isnan(last_value):
+            _x.append(length)
+            _y.append(last_value)
+
+        if len(_y):
+            integral += sum(_y)*(_x[-1]-_x[0])/len(_y)
+        x.append(t)
+        y.append(integral)
+
+    return x, y
```
Comment (PeterPetrik) [inline crayfish/plot.py]: the function should be unit-tested

**Comment by saberraz** [comment]
I get an error with my data:

```
An error has occurred while executing Python code: 

IndexError: list index out of range 
Traceback (most recent call last):
  File "/home/saber/.local/share/QGIS/QGIS3/profiles/default/python/plugins/crayfish/gui/plot_widget.py", line 278, in on_geometries_changed
    self.refresh_plot()
  File "/home/saber/.local/share/QGIS/QGIS3/profiles/default/python/plugins/crayfish/gui/plot_widget.py", line 311, in refresh_plot
    self.refresh_integral_plot()
  File "/home/saber/.local/share/QGIS/QGIS3/profiles/default/python/plugins/crayfish/gui/plot_widget.py", line 427, in refresh_integral_plot
    unit = split[1].split(']')[0]
IndexError: list index out of range
```

**Comment by saberraz** [comment]
crayfish_problem

**Comment by saberraz** [comment]
1- Add a 2dm file
2- Attach the xdmf
3- Plot from Crayfish plugin
4- Select Time Series > From Map (Line)
5- As soon as you click for the first vertex of your line it comes up with python error
6- Ignore the error and keep add vertex for your line
7- Now switch to Cross Section
8- All the vertex you have added in 5 and 6 are already there as a part of cross section

**Comment by saberraz** [comment]
LGTM. Works fine for me

**Pull request merged**